segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/NoOpChunkStorageProvider.java

```diff
@@ -0,0 +1,200 @@
+
+
+    @Override
+    protected ChunkInfo doGetInfo(String chunkName) throws IOException, IllegalArgumentException {
+        ChunkData chunkData = chunkMetadata.get(chunkName);
+        if (null == chunkData) {
+            throw new FileNotFoundException(chunkName);
+        }
+        return ChunkInfo.builder().name(chunkName).length(chunkData.length).build();
+    }
+
+    @Override
+    protected ChunkHandle doCreate(String chunkName) throws IOException, IllegalArgumentException {
+        ChunkData chunkData = chunkMetadata.get(chunkName);
+        if (null != chunkData) {
+            throw new FileAlreadyExistsException(chunkName);
+        }
+        chunkMetadata.put(chunkName, new ChunkData());
+        return ChunkHandle.writeHandle(chunkName);
+    }
+
+    @Override
+    protected boolean doesExist(String chunkName) throws IOException, IllegalArgumentException {
+        return chunkMetadata.containsKey(chunkName);
+    }
+
+    @Override
+    protected boolean doDelete(ChunkHandle handle) throws IOException, IllegalArgumentException {
+        Preconditions.checkNotNull(null != handle, "handle");
+        Preconditions.checkNotNull(handle.getChunkName(), "handle");
+        ChunkData chunkData = chunkMetadata.get(handle.getChunkName());
+        if (null == chunkData) {
+            throw new FileNotFoundException(handle.getChunkName());
+        }
+        if (chunkData.isReadonly) {
+            throw new IOException("chunk is readonly");
+        }
+        chunkMetadata.remove(handle.getChunkName());
+        return true;
+    }
+
+    @Override
+    protected ChunkHandle doOpenRead(String chunkName) throws IOException, IllegalArgumentException {
+        ChunkData chunkData = chunkMetadata.get(chunkName);
+        if (null == chunkData) {
+            throw new FileNotFoundException(chunkName);
+        }
+        return ChunkHandle.readHandle(chunkName);
+    }
+
+    @Override
+    protected ChunkHandle doOpenWrite(String chunkName) throws IOException, IllegalArgumentException {
+        ChunkData chunkData = chunkMetadata.get(chunkName);
+        if (null == chunkData) {
+            throw new FileNotFoundException(chunkName);
+        }
+        return ChunkHandle.writeHandle(chunkName);
+    }
+
+    @Override
+    protected int doRead(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) throws IOException, NullPointerException, IndexOutOfBoundsException {
+        // TODO: Add validation
+        ChunkData chunkData = chunkMetadata.get(handle.getChunkName());
+        if (null == chunkData) {
+            throw new FileNotFoundException(handle.getChunkName());
+        }
+
+        if (fromOffset >= chunkData.length || fromOffset + length > chunkData.length) {
+            throw new IndexOutOfBoundsException("fromOffset");
+        }
+
+        if (fromOffset < 0 || bufferOffset < 0 || length < 0 || buffer.length < bufferOffset + length) {
```

Comment:
Use the appropriate method in `Preconditions` or `Exceptions`.

Reply:
Th exceptions thrown by those methods are slightly different.